test/extended/kubevirt/nodereboot.go

```diff
@@ -0,0 +1,85 @@
+package kubevirt
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	k8sapierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = Describe("[sig-kubevirt] node reboot", func() {
+	oc := exutil.NewCLIWithPodSecurityLevel("ns-global", admissionapi.LevelBaseline)
+
+	InKubeVirtClusterContext(oc, func() {
+		mgmtFramework := e2e.NewDefaultFramework("mgmt-framework")
+		mgmtFramework.SkipNamespaceCreation = true
+
+		hostedFramework := e2e.NewDefaultFramework("hosted-framework")
+		hostedFramework.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+		When("an hosted control plane worker is rebooted", func() {
+			var hostedClusterNodeToBeRebooted corev1.Node
+
+			isNodeReady := func(nodeName string) (bool, error) {
+				node, err := hostedFramework.ClientSet.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+				if k8sapierrors.IsNotFound(err) {
+					return false, err
+				}
```

Comment:
What if it's another error? It will break the rest of the code.

Reply:
I assume the not found node as a non-retrieable error, while others (which I pretty much don't know what they might be) as retriable.

Reply:
I think you will get nil pointer exception in this case, and I'm pretty sure it won't be recoverable for the next loop.

Reply:
could you elaborate where would I get the nil pointer ?

Reply:
Better now ?